Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Isaac Hernando, Coleton Kotch, Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Abelian.Basic
+import Mathlib.CategoryTheory.Limits.Filtered
+import Mathlib.CategoryTheory.Limits.FunctorCategory
+import Mathlib.CategoryTheory.Adjunction.Limits
+
+/-!
+
+# Grothendieck Axioms
+
+This file defines some of the Grothendieck Axioms for abelian categories, and proves
+basic facts about them.
+
+## Definitions
+
+1. `AB4` -- an abelian category satisfies `AB4` provided that coproducts are exact.
+2. `AB5` -- an abelian category satisfies `AB5` provided that filtered colimits are exact.
+
+## Results
+
+- `AB5` implies `AB4`
+
+## Implementation Details
+
+For `AB4` and `AB5`, we only assume left exactness as right exactness is automatic.
+
+## Projects
+
+- Add additional axioms.
+
+-/
+
+namespace CategoryTheory
+
+open Limits Classical
+
+universe v v' u u'
+
+variable (C : Type u) [Category.{v} C]
+
+--Right exactness for arbitrary shapes.
+noncomputable
+example (J : Type u') [Category.{v'} J] [HasColimitsOfShape J C] :
+    PreservesFiniteColimits (colim (J := J) (C := C)) :=
+  inferInstance
+
+/--
+A category `C` which has coproducts is said to have `AB4` provided that
+coproducts are exact.
+-/
+class AB4 [HasCoproducts C] where
+  preservesFiniteLimits (α : Type v) :
+    PreservesFiniteLimits (colim (J := Discrete α) (C := C))
+
+instance [HasCoproducts C] [AB4 C] (α : Type v) :
+    PreservesFiniteLimits (colim (J := Discrete α) (C := C)) :=
+  AB4.preservesFiniteLimits _
+
+/--
+A category `C` which has filtered colimits is said to have `AB5` provided that
+filtered colimits are exact.
+-/
+class AB5 [HasFilteredColimits C] where
+  preservesFiniteLimits (J : Type v) [SmallCategory J] [IsFiltered J] :
+    PreservesFiniteLimits (colim (J := J) (C := C))
+
+instance [HasFilteredColimits C] [AB5 C]
+    (J : Type v) [SmallCategory J] [IsFiltered J] :
+    PreservesFiniteLimits (colim (J := J) (C := C)) :=
+  AB5.preservesFiniteLimits _
+
+variable {C}
+variable {α : Type v} (X Y : α → C) [HasCoproduct X] [HasCoproduct Y]
+variable [HasZeroMorphisms C] [HasFiniteBiproducts C]
+
+/--
+This is the functor `Finset α ⥤ C` which, on objects, sends a finite set `S` to
+`⨁ fun s : S => X s`. We will show below that the colimit of this functor is
+isomorphic to `∐ X`.
+-/
+@[simps]
+noncomputable
+def finsetBiproductDiagram : Finset α ⥤ C where
+  obj S := ⨁ fun s : S => X s
```

Comment:
Can this be written as
```suggestion
  obj S := ⨁ X
```
like the ` ∐ X` below?

Reply:
It's not the same, since we're restricting here to a finite subset of alpha.

Reply:
Oh my bad thanks!